debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -620,6 +620,7 @@ protected void execute() {
                 }
                 step++;
             } finally {
+                mysql.connection().commit();
```

Comment:
This looks odd here, given the rollback attempt below (and second `commit()` in the else branch). Can this just be removed?

Reply:
I've changed the `commit` to `rollback` and placed it to a separata branch to cover all potenital cases.